File: play/tmp/encrypt.js
function encryptCustomKey(beforeKey) {
    var b_key = [];
    var r_key = new Array(16);

    for (var i = 0; i < beforeKey.length; ++i) {
        b_key.push(beforeKey.charCodeAt(i));
    }

    for (var i = 0; i < 8; ++i) {
        var temp = (b_key[i] ^ 0xfef0) % 100 + 27;
        if (temp - 32 <= 0) {
            temp += 32;
        }
        r_key[i] = (b_key[15 - i] ^ 0xfabc) % 100 + 27;
        if (r_key[i] - 32 <= 0) {
            r_key[i] += 32;
        }
        r_key[15 - i] = temp;
    }
    return String.fromCharCode(...r_key);
}

var res = encryptCustomKey("1234一二三四abcd😋😋😋😋");
console.log(res);

File: play/tmp/Encrypt.java
public class Encrypt {
    public static String encryptCustomKey(String beforeKey) {
        char[] b_key = beforeKey.toCharArray();
        char[] r_key = new char[16];
        for (int i = 0; i < 8; ++i) {
            char temp = (char) ((b_key[i] ^ 0xfef0) % 100 + 27);
            if (temp - 32 <= 0) {
                temp += 32;
            }
            r_key[i] = (char) ((b_key[15 - i] ^ 0xfabc) % 100 + 27);
            if (r_key[i] - 32 <= 0) {
                r_key[i] += 32;
            }
            r_key[15 - i] = temp;
        }

        return String.valueOf(r_key);
    }

    public static void main(String[] args) {
        String s = "1一😋فور 하나.ทั้งมวล二三四234";
        System.out.println(encryptCustomKey(s));
    }
}
